docs/groovy-script/mods/academy/metal_former.md

```diff
@@ -0,0 +1,145 @@
+---
+title: "Metal Former"
+titleTemplate: "AcademyCraft | CleanroomMC"
+description: "Converts an item into another item, according to the mode of the machine."
+source_code_link: "https://github.com/yor42/AcademyCraft/blob/master/src/main/java/cn/academy/support/groovyscript/modules/MetalFormer.java"
+---
+
+# Metal Former (AcademyCraft)
+
+## Description
+
+Converts an item into another item, according to the mode of the machine.
+
+## Identifier
+
+Refer to this via any of the following:
+
+```groovy:no-line-numbers {1}
+mods.academy.metal_former/* Used as page default */ // [!code focus]
+mods.academy.metalformer
+mods.academy.metalFormer
+mods.academy.MetalFormer
+```
+
+
+## Adding Recipes
+
+- Add the given recipe to the recipe list:
+
+    ```groovy:no-line-numbers
+    mods.academy.metal_former.add(MetalFormerRecipes.RecipeObject)
+    ```
+
+- Adds etching recipes in the format `input`, `output`:
+
+    ```groovy:no-line-numbers
+    mods.academy.metal_former.addEtchRecipe(IIngredient, ItemStack)
+    ```
+
+- Adds incising recipes in the format `input`, `output`:
+
+    ```groovy:no-line-numbers
+    mods.academy.metal_former.addInciseRecipe(IIngredient, ItemStack)
+    ```
+
+- Adds plating recipes in the format `input`, `output`:
+
+    ```groovy:no-line-numbers
+    mods.academy.metal_former.addPlateRecipe(IIngredient, ItemStack)
+    ```
+
+- Adds refining recipes in the format `input`, `output`:
+
+    ```groovy:no-line-numbers
+    mods.academy.metal_former.addRefineRecipe(IIngredient, ItemStack)
+    ```
+
+
+## Removing Recipes
+
+- Removes the given recipe from the recipe list:
+
+    ```groovy:no-line-numbers
+    mods.academy.metal_former.remove(MetalFormerRecipes.RecipeObject)
+    ```
+
+- Removes all etching entry of metal former:
+
+    ```groovy:no-line-numbers
+    mods.academy.metal_former.clearEtch()
+    ```
+
+- Removes all etching entry of metal former:
+
+    ```groovy:no-line-numbers
+    mods.academy.metal_former.clearIncise()
+    ```
+
+- Removes all etching entry of metal former:
+
+    ```groovy:no-line-numbers
+    mods.academy.metal_former.clearPlate()
+    ```
+
+- Removes all entry of metal former:
+
+    ```groovy:no-line-numbers
+    mods.academy.metal_former.clearRecipe()
+    ```
+
+- Removes all etching entry of metal former:
+
+    ```groovy:no-line-numbers
+    mods.academy.metal_former.clearRefine()
+    ```
+
+- Removes an entry matching the given `ItemStack` from etching recipes:
+
+    ```groovy:no-line-numbers
+    mods.academy.metal_former.removeEtchbyInput(ItemStack)
```

Comment:
by standard java method capitalization rules, this should be `removeEtchByInput` - the `B` in `By` is lowercase here.